casper/src/main/rholang/WalletCheck.rho

```diff
@@ -31,9 +31,9 @@ contract @["WalletCheck", "create"](@hash, @purse) = {
                 new hashOut in {
 
                   // Check that given pubKey is the preimage of hash
-                  @"keccak256Hash"!(pubKey.toByteArray(), *hashOut) |
+                  algorithm!(pubKey.toByteArray(), *hashOut) |
                   for (@pkHash <- hashOut) {
-                    if (pkHash == hash) {
+                    if (pkHash == hash.hexToBytes()) {
```

Comment:
You should update the type annotation in your comment if you are going to now treat `hash` like a string.